org.eclipse.m2e.profiles.ui/src/org/eclipse/m2e/profiles/ui/internal/actions/ProfileSelectionHandler.java

```diff
@@ -72,9 +78,19 @@ public class ProfileSelectionHandler extends AbstractHandler {
 
   private static final Logger log = LoggerFactory.getLogger(ProfileSelectionHandler.class);
 
+  @Inject
+  private IProfileManager profileManager;
+
+  public ProfileSelectionHandler() {
+    BundleContext context = FrameworkUtil.getBundle(ProfileSelectionHandler.class).getBundleContext();
```

Comment:
This plug-ins manifest specifies `Bundle-ActivationPolicy: lazy` so the `BundleContext` should never be null here.
Furthermore I think this code will never run outside of OSGi so I also expect that `getBundle()` never returns null unless the entire framework is in a weird state.

Reply:
This still do not handle dynamics, we should not use the bare OSGi API when there is no need for it and we can use a DI technology instead.

Reply:
What's then the approach you suggest?

Reply:
Sorry for confusion ... it seems it get messed up with the recent push, as long as Eclipse offers not the `EclipseContextFactory.getServiceContext(<class>);` method think we can go with this...

Reply:
> Sorry for confusion ... it seems it get messed up with the recent push, as long as Eclipse offers not the `EclipseContextFactory.getServiceContext(<class>);` method think we can go with this...

No problem. Makes sense to use that ones available.